spec/fortnox/api/types/country_code_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'fortnox/api/types'
+require 'fortnox/api/types/examples/types'
+
+describe Fortnox::API::Types::CountryCode do
+  context 'with nil' do
+    subject { described_class[nil] }
+
+    it { is_expected.to be_nil }
+  end
+
+  context 'with empty string' do
+    subject { described_class[''] }
+
+    it { is_expected.to eq('') }
+  end
+
+  context 'with valid input' do
+    it 'accepts country codes' do
+      expect(described_class['NO']).to eq 'NO'
+    end
+
+    it 'translates English country names to country code' do
+      expect(described_class['Norway']).to eq 'NO'
+    end
+
+    it 'translates Swedish country names to country code' do
+      expect(described_class['Norge']).to eq 'NO'
+    end
+
+    describe 'special cases' do
+      valid_sweden_inputs = [
+        'SE', 'se', 'Sweden', 'sweden', 'Sverige', 'sverige',
+        :SE, :se, :Sweden, :sweden, :Sverige, :sverige
+      ].freeze
+
+      valid_sweden_inputs.each do |sweden_input|
+        it "converts \"#{sweden_input}\" to \"SE\"" do
+          expect(described_class[sweden_input]).to eq 'SE'
+        end
+      end
+    end
+  end
+
+  context 'with invalid input' do
+    invalid_inputs = [
+      'SEA', 'S', 'nonsense', :s
```

Comment:
Ah, this is where you should test with `nil` and empty string then? They are special cases in the implementation so ...

Reply:
Great that you pointed this out! But both `nil` and empty string is valid input, so they should be tested above in the code :)